solidity-v1/dashboard/src/components/threshold/AllocationProgressBar.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useMemo } from "react"
+import ProgressBar, {
+  ProgressBarLegendContext,
+  renderProgressBarLegendItem,
+} from "../ProgressBar"
+import { colors } from "../../constants/colors"
+import OnlyIf from "../OnlyIf"
+import { add, gt } from "../../utils/arithmetics.utils"
+
+const AllocationProgressBar = ({
+  title,
+  currentValue = 0,
+  totalValue = 100,
+  className = "",
+  secondaryValue = null,
+  withLegend = false,
+  currentValueLegendLabel = "",
+  secondaryValueLegendLabel = "",
+  isDataFetching = false,
+}) => {
+  const percentageValue = useMemo(() => {
+    if (isDataFetching) {
+      return "--"
+    }
+
+    if (gt(totalValue, 0)) {
+      return Math.round(
+        (add(currentValue, secondaryValue | 0) / totalValue) * 100
+      )
+    } else {
+      return 0
+    }
+  }, [currentValue, secondaryValue, totalValue, isDataFetching])
+
+  return (
+    <div className={`allocation-progress-bar ${className}`}>
+      <div>
+        <h5 className="text-grey-60">{title}</h5>
+        <div className="allocation-progress-bar__progress-bar-container">
+          <ProgressBar
+            total={isDataFetching ? 100 : totalValue}
+            bgColor={colors.grey20}
+          >
+            <ProgressBar.Inline
+              className="allocation-progress-bar__progress-bar-wrapper"
+              height={20}
+            >
+              <ProgressBar.InlineItem
+                value={
+                  isDataFetching ? 0 : add(currentValue, secondaryValue | 0)
+                }
+                color={colors.secondary}
+              />
+              <OnlyIf condition={!!secondaryValue}>
+                <ProgressBar.InlineItem
+                  value={isDataFetching ? 0 : secondaryValue}
+                  color={colors.yellowSecondary}
+                />
+              </OnlyIf>
+            </ProgressBar.Inline>
+            <OnlyIf condition={withLegend}>
+              <div className="allocation-progress-bar__progress-bar-legend">
+                <ProgressBarLegendContext.Provider
+                  value={{ renderValuePattern: () => {} }}
```

Comment:
Can we render an empty component? I mean `() => <></>`.

Reply:
Fixed in 62510cdb6fc522e3af4414f7563b1916c0a69571